### PR TITLE
fix(pie-monorepo): DSW-1376 mark "lit/react" as an external dependency 

### PR DESCRIPTION
**Pull request by raoufswe**
## Describe your changes (can list changeset entries if preferable)

- This PR marks "@lit/react" as an external package so it isn't bundled as part of the react wrappers. 
- This is required to fix (DSW-1376) where the react-wrappers of our component will fall back to their default values when JS is disabled resulting in a rehydration error. 
- The lit team also suggested the package should be marked as `external` to avoid other edge cases. 
- the change was tested in all of our react examples and should be good to go 🥳 

the lit team also suggested to not bundle the package as that will just create duplications for consumers.
## Author Checklist (complete before requesting a review)
- [ ] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests
- [ ] If it is a `PIE Docs` change, I have reviewed the Docs site preview
- [ ] If it is a component change, I have reviewed the Storybook preview
- [ ] If there are visual test updates, I have reviewed them properly before approving

### Reviewer 1
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### Reviewer 2
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### CODE REVIEW

File: configs/pie-components-config/vite.config.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ const sharedConfig = ({ build = {}, plugins = [], ...rest }) => defineConfig({
         },
         rollupOptions: {
             external: (id) => {
-                if (id === 'react' || /^lit/.test(id)) {
+                if (id === 'react' || /^lit/.test(id) || id === '@lit/react') {
```
Comment (raoufswe) [inline configs/pie-components-config/vite.config.js:25]: this is the main change, the rest are just readme updates to use the new package name :D

File: packages/tools/pie-wrapper-react/CHANGELOG.md [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 ### Minor Changes
 
-- [Changed] - `@lit-labs/react` dependency to `@lit/react` and updated any references. ([#999](https://github.com/justeattakeaway/pie/pull/999)) by [@siggerzz](https://github.com/siggerzz)
+- [Changed] - `@lit/react` dependency to `@lit/react` and updated any references. ([#999](https://github.com/justeattakeaway/pie/pull/999)) by [@siggerzz](https://github.com/siggerzz)
```
Comment (fernandofranca) [inline packages/tools/pie-wrapper-react/CHANGELOG.md:7]: I believe this one should still be `@lit-labs/react`

Comment (raoufswe) [reply packages/tools/pie-wrapper-react/CHANGELOG.md:7]: fixed 💯

File: configs/pie-components-config/vite.config.js (hunk 1 above)
Comment (kevinrodrigues) [reply configs/pie-components-config/vite.config.js:25]: Not tested but i think you can shorten this a bit to make it a bit more readable:

`['react', '@lit/react'].includes(id) || /^lit/.test(id)`

**Review by kevinrodrigues — APPROVED** [review]

File: packages/components/pie-modal/README.md [hunk 3]
```diff
@@ -57,7 +57,7 @@ import '@justeattakeaway/pie-modal';
 ```js
 // React
 // For React, you will need to import our React-specific component build
-// which wraps the web component using @lit-labs/react
+// which wraps the web component using @lit/react
```
Comment (xander-marjoram) [inline packages/components/pie-modal/README.md:60]: I'm wondering whether we need this second line at all. Do consumers need to know that we use `@lit/react`?

Comment (raoufswe) [reply packages/components/pie-modal/README.md:60]: I am not sure, I would assume it was added for a reason.

File: configs/pie-components-config/vite.config.js (hunk 1 above)
Comment (raoufswe) [reply configs/pie-components-config/vite.config.js:25]: I've thought about that but `includes` isn't strict and one of the id entries is the `react.ts` file which shouldn't be marked as external.

Comment (raoufswe) [reply configs/pie-components-config/vite.config.js:25]: ahh yea makes sense

File: packages/components/pie-modal/README.md (hunk 3 above)
Comment (raoufswe) [reply packages/components/pie-modal/README.md:60]: we agreed to address this in a follow-up PR as the react section of all components is misleading.

**Review by xander-marjoram — APPROVED** [review]

**Pull request merged**